src/pages/Interferometry/MachZehnderEquipment.js

```diff
@@ -50,8 +113,34 @@ class MachZehnderEquipment {
     drawThePhotonPath(context, this.mirror1, this.bs2);
     drawThePhotonPath(context, this.bs2, this.detector0);
     drawThePhotonPath(context, this.bs2, this.detector1);
+  }
+
+  fire(context, photon, photonFire, probability) {
+    if (photonFire) {
+      photon.move(this.detector0, this.mirror0, this.bs1, this.bs2, probability);
+      photon.drawParticle(context);
 
-    context.restore();
+      if (photon.posX > this.detector0.posX && photon.posY < this.detector0.posY) {
+        context.fillRect(this.detector0.posX - 60, this.detector0.posY, 30, 5);
+        context.fillRect(this.detector0.posX, this.detector0.posY + 30, 5, 30);
+      }
+      if (photon.posX > this.detector1.posX && photon.posY > this.detector1.posY) {
+        context.fillRect(this.detector1.posX - 60, this.detector1.posY, 30, 5);
+        context.fillRect(this.detector1.posX, this.detector1.posY - 60, 5, 30);
+      }
+      if (photon.posX > this.detector0.posX) {
+        if (photon.posY < this.detector0.posY) { // click detector 0
+          this.countsD0 += 1;
+          this.counts += 1;
+        } else { // click detector 1
+          this.countsD1 += 1;
+          this.counts += 1;
```

Comment:
`this.counts += 1;` should be outside of this if-else because these both if and else clauses have same line.